crates/astria-bridge-withdrawer/src/withdrawer/submitter/mod.rs

```diff
@@ -0,0 +1,269 @@
+            Err(reason) => {
+                error!(%reason, "starting shutdown process")
+            }
+        }
+
+        // handle shutdown
+
+        Ok(())
+    }
+}
+
+/// Queries the sequencer for the latest nonce with an exponential backoff
+#[instrument(name = "get latest nonce", skip_all, fields(%address))]
+async fn get_latest_nonce(
+    client: sequencer_client::HttpClient,
+    address: Address,
+    state: Arc<State>,
+) -> eyre::Result<u32> {
+    debug!("fetching latest nonce from sequencer");
+    metrics::counter!(crate::metrics_init::NONCE_FETCH_COUNT).increment(1);
+    let span = Span::current();
+    let start = Instant::now();
+    let retry_config = tryhard::RetryFutureConfig::new(1024)
+        .exponential_backoff(Duration::from_millis(200))
+        .max_delay(Duration::from_secs(60))
+        .on_retry(
+            |attempt,
+             next_delay: Option<Duration>,
+             err: &sequencer_client::extension_trait::Error| {
+                metrics::counter!(crate::metrics_init::NONCE_FETCH_FAILURE_COUNT).increment(1);
+
+                let state = Arc::clone(&state);
+                state.set_sequencer_connected(false);
+
+                let wait_duration = next_delay
+                    .map(humantime::format_duration)
+                    .map(tracing::field::display);
+                warn!(
+                    parent: span.clone(),
+                    error = err as &dyn std::error::Error,
+                    attempt,
+                    wait_duration,
+                    "failed getting latest nonce from sequencer; retrying after backoff",
+                );
+                async move {}
+            },
+        );
+    let res = tryhard::retry_fn(|| {
+        let client = client.clone();
+        let span = info_span!(parent: span.clone(), "attempt get nonce");
+        async move { client.get_latest_nonce(address).await.map(|rsp| rsp.nonce) }.instrument(span)
+    })
+    .with_config(retry_config)
+    .await
+    .wrap_err("failed getting latest nonce from sequencer after 1024 attempts");
+
+    state.set_sequencer_connected(res.is_ok());
+
+    metrics::histogram!(crate::metrics_init::NONCE_FETCH_LATENCY).record(start.elapsed());
+
+    res
+}
+
+/// Queries the sequencer for the latest nonce with an exponential backoff
+#[instrument(
+    name = "submit signed transaction",
+    skip_all,
+    fields(
+        nonce = tx.unsigned_transaction().params.nonce,
+        transaction.hash = hex::encode(sha256(&tx.to_raw().encode_to_vec())),
+    )
+)]
+async fn submit_tx(
+    client: sequencer_client::HttpClient,
+    tx: SignedTransaction,
+    state: Arc<State>,
+) -> eyre::Result<tx_commit::Response> {
+    let nonce = tx.unsigned_transaction().params.nonce;
+    metrics::gauge!(crate::metrics_init::CURRENT_NONCE).set(nonce);
+
+    // TODO: change to info and log tx hash (to match info log in `SubmitFut`'s response handling
+    // logic)
```

Comment:
change the log below to info? also the tx hash is in the span so don't need to re-add

Reply:
ah missed this when copying over stuff from composer 🤦